PSPs/psp-22.md

```diff
@@ -9,181 +9,674 @@
- /// * Panics with `ZeroSenderAddress` error if sender's address is zero.
- /// * Panics with `ZeroRecipientAddress` error if recipient's address is zero.
- fn decrease_allowance(&mut self, spender: AccountId, delta_value: Balance);
+##### **balance_of**(owner: AccountId) -> Balance
+```json
+{
+  "args": [
+    {
+      "name": "owner",
+      "type": {
+        "displayName": [
+          "AccountId"
+        ],
+        "type": "AccountId"
+      }
+    }
+  ],
+  "docs": [
+    " Returns the account Balance for the specified `owner`.",
+    "",
+    " Returns `0` if the account is non-existent."
+  ],
+  "mutates": false,
+  "name": [
+    "PSP22",
+    "balance_of"
+  ],
+  "payable": false,
+  "returnType": {
+    "displayName": [
+      "Balance"
+    ],
+    "type": "Balance"
+  },
+  "selector": "0x936205de"
 }
+```
 
-/// PSP22Metadata is an optional trait of metadata for Fungible Token Standard.
-pub trait PSP22Metadata {
- /// Returns the token name.
- fn name(&self) -> Option<String>;
+##### **allowance**(owner: AccountId, spender: AccountId) -> Balance
+```json
+{
+  "args": [
+    {
+      "name": "owner",
+      "type": {
+        "displayName": [
+          "AccountId"
+        ],
+        "type": "AccountId"
+      }
+    },
+    {
+      "name": "spender",
+      "type": {
+        "displayName": [
+          "AccountId"
+        ],
+        "type": "AccountId"
+      }
+    }
+  ],
+  "docs": [
+    " Returns the amount which `spender` is still allowed to withdraw from `owner`.",
+    "",
+    " Returns `0` if no allowance has been set `0`."
+  ],
+  "mutates": false,
+  "name": [
+    "PSP22",
+    "allowance"
+  ],
+  "payable": false,
+  "returnType": {
+    "displayName": [
+      "Balance"
+    ],
+    "type": "Balance"
+  },
+  "selector": "0x467e8bff"
+}
+```
 
- /// Returns the token symbol.
- fn symbol(&self) -> Option<String>;
+##### **transfer**(to: AccountId, value: Balance, data: Vec<u8>)
+```json
+{
+  "args": [
+    {
+      "name": "to",
+      "type": {
+        "displayName": [
+          "AccountId"
+        ],
+        "type": "AccountId"
+      }
+    },
+    {
+      "name": "value",
+      "type": {
+        "displayName": [
+          "Balance"
+        ],
+        "type": "Balance"
+      }
+    },
+    {
+      "name": "data",
+      "type": {
+        "displayName": [
+          "Vec"
+        ],
+        "type": "Vec<u8>"
+      }
+    }
+  ],
+  "docs": [
+    " Transfers `value` amount of tokens from the caller's account to account `to`",
+    " with additional `data` in unspecified format..",
+    "",
+    " On success a `Transfer` event is emitted.",
+    "",
+    " # Errors",
+    "",
+    " Panics with message `InsufficientBalance` if there are not enough tokens on",
+    " the caller's account Balance.",
+    "",
+    " Panics with message `ZeroSenderAddress` if sender's address is zero.",
+    "",
+    " Panics with message `ZeroRecipientAddress` if recipient's address is zero."
```

Comment:
remember the comments on panics? maybe we should adapt this as @athei said. like "Returns with REVERT flag with `InsufficientBalance` encoded in the buffer."